tests/jobs/scala/src/main/scala/KerberizedKafkaProducer.scala

```diff
@@ -35,14 +35,28 @@ object KerberizedKafkaProducer {
 
   def sendMessages(topic: String, messages: Array[String])(producer: KafkaProducer[Int, String]): Unit = {
     messages.foreach { str =>
-      val msg = new ProducerRecord[Int, String](topic, str.hashCode, str)
+      val msg = new ProducerRecord[Int, String](topic, str)
       println(s"sending message: $msg")
 
-      Try(producer.send(msg)) match {
-        case Success(_) => println("Message sent")
-        case Failure(ex) => ex.printStackTrace()
+      producer.send(msg, new Callback {
+      override def onCompletion(metadata: RecordMetadata, exception: Exception): Unit = {
+        if (exception != null) {
+          exception.printStackTrace()
+        } else {
+          println(s"Message has been published to (topic_partition@offset): ${metadata.toString}")
+        }
       }
+      })
     }
+    Thread.sleep(3000) // wait a few seconds for metrics update
+    println("===========================================================================")
+    println("Producer metrics by topic:")
+    producer.metrics.values
+      .filter(m => m.metricName().group().equalsIgnoreCase("producer-topic-metrics"))
+      .foreach(metric => println(metric.metricName().description() + ": " + metric.metricValue()))
+    println("===========================================================================")
+    println(s"Partition metadata for topic '$topic':")
```

Comment:
can you please refactor the file and use `Logger` for logging instead of `println`?

Reply:
sure. do you mean all the statements or only the new ones? It seems like it's no-brainer, making object extend `Logger` did the trick, but statements logged with `logInfo()` appeared in stderr log and I guess won't be displayed in test's output. Is it so?

Reply:
good catch! in pytests we grep the output from stdout, let's keep `println` then